packages/terra-application-utility/src/header/ApplicationHeaderUtility.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import IconChevronDown from 'terra-icon/lib/icon/IconChevronDown';
+import UtilityMenu from './_HeaderUtilityMenu';
+import Utils from '../_Utils';
+import styles from './ApplicationHeaderUtility.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The array containing the menu items to be rendered within the menu.
+   */
+  menuItems: PropTypes.arrayOf((Utils.itemShape)).isRequired,
+  /**
+   * The function to trigger when a menu item is selected.
+   */
+  onChange: PropTypes.func.isRequired,
+    /**
+   * The function that discloses the menu.
+   */
+  onDisclose: PropTypes.func.isRequired,
+  /**
+   * The name to be displayed next to the user photo.
+   */
+  userName: PropTypes.string,
+  /**
+   * The image associated with the user.
+   */
+  userPhoto: PropTypes.element,
+};
+
+const defaultProps = {
+  menuItems: [],
+  userName: '',
+  userPhoto: undefined,
+};
+
+class ApplicationHeaderUtility extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleOnClick = this.handleOnClick.bind(this);
+    this.createContent = this.createContent.bind(this);
+  }
+
+  handleOnClick() {
+    if (this.props.onDisclose) {
+      const content = this.createContent();
+      this.props.onDisclose(content);
+    }
+  }
+
+  createContent() {
+    return (
+      <UtilityMenu
+        initialSelectedKey={Utils.KEYS.MENU}
+        menuItems={this.props.menuItems}
+        onChange={this.props.onChange}
+        data-application-header-utility-menu
+      />
+    );
+  }
+
+  render() {
+    const {
+      menuItems,
+      onChange,
+      onDisclose,
+      userName,
+      userPhoto,
+      ...customProps
+    } = this.props;
+
+    const utilityClassNames = cx(['header-utility', customProps.className]);
+    const userPhotoClassNames = cx('user-photo');
+    const userNameClassNames = cx('user-name');
+    const iconClassNames = cx('icon');
+    let photo = null;
+    if (userPhoto) {
+      photo = React.cloneElement(userPhoto, { className: userPhotoClassNames });
+    }
+    return (
+      <button {...customProps} className={utilityClassNames} onClick={this.handleOnClick} role={'main'} aria-label={'Utility button'}>
```

Comment:
Is there a reason we didn't use the terra-button for this piece? There will be differences in keyboard accessibility if we don't include it here.

Reply:
I don't think 'main' is the appropriate role to be adding to this button.

It is noted [here](http://html5doctor.com/the-main-element/) that one should
>use the ARIA role until all browsers map the role to the < main > element. The primary purpose of < main > is to map ARIA’s landmark role main to an element in HTML. This will help screen readers and other assistive technologies understand where the main content begins.

Reply:
`terra-button` doesn't accept children - this button needs the ability to render an image (passed down as a child), along with text and an icon.